src/NServiceBus.Core/Serialization/SerializationConfigExtensions.cs

```diff
@@ -52,6 +53,27 @@ public static void UseSerialization(this BusConfiguration config, Type serialize
             config.Settings.Set("CustomSerializerType", serializerType);
         }
 
+        /// <summary>
+        /// Configures additional deserializers to be considered when processing messages. Can be called multiple times.
+        /// </summary>
+        /// <typeparam name="T">The serializer definition eg <see cref="JsonSerializer"/>, <see cref="XmlSerializer"/>, etc.</typeparam>
+        /// <param name="config">The <see cref="BusConfiguration"/> instance to apply the settings to.</param>
+        public static void AddDeserializer<T>(this BusConfiguration config) where T : SerializationDefinition
+        {
+            Guard.AgainstNull("config", config);
+
+            HashSet<SerializationDefinition> deserializers;
+            if (!config.Settings.TryGet("AdditionalDeserializers", out deserializers))
+            {
+                deserializers = new HashSet<SerializationDefinition>();
+            }
+
+            var instance = Activator.CreateInstance<T>();
+            deserializers.Add(instance);
+
+            config.Settings.Set("AdditionalDeserializers", deserializers);
```

Comment:
just set inside the `if (!config.Settings.` to save re setting if it already exists